libs/components/core/src/lib/modules/dynamic-component/dynamic-component.service.ts

```diff
@@ -67,53 +67,60 @@ export class SkyDynamicComponentService {
       parent: options.parentInjector || this.#injector,
     });
 
-    const componentRef = createComponent<T>(componentType, {
-      environmentInjector: this.#applicationRef.injector,
-      elementInjector: injector,
-    });
-
-    this.#applicationRef.attachView(componentRef.hostView);
-
-    const el = this.#getRootNode(componentRef);
-
-    const bodyEl = this.#windowRef.nativeWindow.document.body;
-
-    switch (options.location) {
-      case SkyDynamicComponentLocation.BeforeElement:
-        if (!options.referenceEl) {
-          throw new Error(
-            '[SkyDynamicComponentService] Could not create a component at location `SkyDynamicComponentLocation.BeforeElement` because a reference element was not provided.'
+    let componentRef: ComponentRef<T>;
+    if (options.viewRef) {
+      componentRef = options.viewRef.createComponent(componentType, {
+        injector: injector,
+      });
+    } else {
+      componentRef = createComponent<T>(componentType, {
```

Comment:
Before we merge, can we figure out why `createComponent` wasn't working? What do `environmentInjector` and `elementInjector` do?

Reply:
`createComponent` seems not to want to create a new element with the injection scope we are attempting to pass in. `environmentInjector` is essentially a top level scope, and `elementInjector` is an optional localized one, but for some reason passing this injector in doesn't contain a reference to a service provided at the parent component level.